src/write.rs

```diff
@@ -133,6 +134,13 @@ impl<W: Write + io::Seek> Write for ZipWriter<W> {
                 "No file has been started",
             ));
         }
+
+        // Check if the buffer is text (UTF8) or contains binary data. Note that if an UTF8
+        // character is split between to calls to write, this will falsly mark it as binary.
+        // For ASCII characters, this is always correct.
+        if self.stats.is_text == true && std::str::from_utf8(buf).is_err() {
+            self.stats.is_text = false;
+        }
```

Comment:
This behavior seems annoying - if the user has indicated that they're *trying* to create a text file, they probably don't want it to just silently fail. 

I'd like to investigate using `fmt::Write` to require the user to validate the input (`from_utf8_unchecked` when they want to be quick), but maybe there's a better solution.

For now, can we just return an error here instead? *something* is wrong with the input we've been given, and the user should know